chainio/txmgr/geometric/geometric.go

```diff
@@ -0,0 +1,609 @@
+	ConfirmationBlocks int
+	// time to wait for a transaction to be broadcasted to the network
+	// could be direct via eth_sendRawTransaction or indirect via a wallet service such as fireblocks
+	// default: 2 minutes
+	TxnBroadcastTimeout time.Duration
+	// time to wait for a transaction to be confirmed (mined + confirmationBlocks blocks)
+	// default: 5 * 12 seconds
+	TxnConfirmationTimeout time.Duration
+	// max number of times to retry sending a transaction before failing
+	// this applies to every transaction attempt when a nonce is bumped
+	// default: 3
+	MaxSendTransactionRetry int
+	// time to wait between checking for each transaction receipt
+	// while monitoring transactions to get mined
+	// default: 3 seconds
+	GetTxReceiptTickerDuration time.Duration
+	// default gas tip cap to use when eth_maxPriorityFeePerGas is not available
+	// default: 5 gwei
+	FallbackGasTipCap uint64
+	// multiplier for gas limit to add a buffer and increase chance of tx getting included. Should be >= 1.0
+	// default: 1.2
+	GasMultiplier float64
+	// multiplier for gas tip. Should be >= 1.0
+	// default: 1.25
+	GasTipMultiplier float64
+}
+
+var defaultParams = GeometricTxnManagerParams{
+	ConfirmationBlocks:         0,                    // tx mined is considered confirmed
+	TxnBroadcastTimeout:        2 * time.Minute,      // fireblocks has had issues so we give it a long time
+	TxnConfirmationTimeout:     5 * 12 * time.Second, // 5 blocks
+	MaxSendTransactionRetry:    3,                    // arbitrary
+	GetTxReceiptTickerDuration: 3 * time.Second,
+	FallbackGasTipCap:          uint64(5_000_000_000), // 5 gwei
+	GasMultiplier:              float64(1.20),         // add an extra 20% gas buffer to the gas limit
+	GasTipMultiplier:           float64(1.25),         // add an extra 25% to the gas tip
+}
+
+func fillParamsWithDefaultValues(params *GeometricTxnManagerParams) {
+	if params.ConfirmationBlocks == 0 {
+		params.ConfirmationBlocks = defaultParams.ConfirmationBlocks
+	}
+	if params.TxnBroadcastTimeout == 0 {
+		params.TxnBroadcastTimeout = defaultParams.TxnBroadcastTimeout
+	}
+	if params.TxnConfirmationTimeout == 0 {
+		params.TxnConfirmationTimeout = defaultParams.TxnConfirmationTimeout
+	}
+	if params.MaxSendTransactionRetry == 0 {
+		params.MaxSendTransactionRetry = defaultParams.MaxSendTransactionRetry
+	}
+	if params.GetTxReceiptTickerDuration == 0 {
+		params.GetTxReceiptTickerDuration = defaultParams.GetTxReceiptTickerDuration
+	}
+	if params.FallbackGasTipCap == 0 {
+		params.FallbackGasTipCap = defaultParams.FallbackGasTipCap
+	}
+	if params.GasMultiplier == 0 {
+		params.GasMultiplier = defaultParams.GasMultiplier
+	}
+	if params.GasTipMultiplier == 0 {
+		params.GasTipMultiplier = defaultParams.GasTipMultiplier
+	}
+}
+
+func NewGeometricTxnManager(
+	ethClient ethBackend,
+	wallet wallet.Wallet,
+	logger logging.Logger,
+	metrics Metrics,
+	params GeometricTxnManagerParams,
+) *GeometricTxManager {
+	fillParamsWithDefaultValues(&params)
```

Comment:
Not sure if enforcing the default values here makes the most sense. 
How about providing the default params as exported value so callers can use it, and just validating these params in the constructor (returning error if any params missing)?

Reply:
They are not enforced! They are only set when the value is unset (has the zero value, which doesnt make sense for all params). Changed the name to reflect that: [09d3e23](https://github.com/Layr-Labs/eigensdk-go/pull/284/commits/09d3e23a483cbe3e5d5081b09a2113c71f20c4ca)

Reply:
What I meant was, the constructor sets the params under the hood (if some values are missing) and the caller may not realize that. Was thinking it would be better to be explicit

Reply:
I don't think so. It's very typical to have a bunch of optional parameters with defaults. aws sdk has it for eg. One nice way to set them is with the functional options pattern: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis.
But if there's a lot of parameters like here, it's just very annoying to have to set all of them explicitly, especially when you don't know what to set. And having to write 10 default params is just annoying I feel.